src/lib/Libattr/master_sched_attr_def.xml

```diff
@@ -258,7 +258,7 @@
         <member_at_comp>comp_l</member_at_comp>
         <member_at_free>free_null</member_at_free>
         <member_at_action>action_sched_port</member_at_action>
-        <member_at_flags><both>MGR_ONLY_SET</both></member_at_flags>
+        <member_at_flags><both>READ_ONLY | ATR_DFLAG_SSET</both></member_at_flags>
```

Comment:
@arungrover / @bhroam 
Let me explain what I have implemented with respect to the scheduler updating its attributes like sched_port, sched_host etc to PBS Server. 
* As per the initial design, as a step 1. admin can select a port for the new scheduler, step 2. if this 
  port is not free then scheduler anyway picks up the free port available. 
* Thought of reducing one step for the administrator in such a way that scheduler(other than the 
  default scheduler) itself picks-up one of the available ports starting from 15050.  Wanted to 
  update the document only after reviewers also like this idea.
* So In order to meet the above requirement the new scheduler itself should find out the free 
   port and also has to update this back to the server which means it has to do the following things.
  * Very first time, scheduler should act as a client and then connect to the server. Here I am 
     calling the function cnt2server_extend() to get the connector fd. 
   * It has to update server about the selected port by calling update_svr_schedobj(). In this 
     function pbs_manager() gets called which at least requires manager privileges to accomplish 
     the goal.  At the same time I also wanted to make sure that manager or operator can not 
     change the port. To meet both these requirements at the same time,  I am passing 
     SC_DAEMON in extend parameter for  cnt2server_extend(). This is useful in identifying that the request is from the scheduler(who is also manager and its privilege is beyond that), so others can not modify it. (For the server side code changes please look in to req_connect() function) 
   * After updating the port to the server, Immediately calling pbs_disconnect(connector) to 
     disconnect it as client.  Then scheduler becomes like a server as usual.
   * Even though scheduler acts as a client only for the first time due to the above code changes we are also making sure that the request is actually from the scheduler through which we are taking care of authentication issues as well. Also after the next interface of Multisched feature PBS Server is the one who actually controls scheduler i.e. starting, stopping etc. So this level of authentication is enough as per my knowledge.
* One more approach is admin himself has to find the free port available on the host where scheduler runs and configure the port through qmgr.  If the port is not available then the scheduler won't come up. He again has to select one more port and configure it through qmgr.

Please suggest if you have a better solution other than the above approaches so that I can follow it.

Reply:
@suresh-thelkar Now when I think about it, I find it problematic if scheduler decides the port and reports that back to server. We will have to handle a lot of error conditions and sync issues between the time scheduler object is created and scheduler starts and report back the port number. I like your second option better where scheduler will only start on an admin specified port, it is more precise that way.

Reply:
@bhroam 
As @arungrover likes the second approach, If you are also fine with it then I can go ahead and implement it accordingly. This decision helps me in unblocking/addressing some of the other review comments also.

Reply:
I agree with Arun about the second approach.  In your next pull request, you will have the server starting schedulers.  When that happens, the server will be telling the scheduler what port to start on.  You'd have to rip out all of this code in the scheduler (not to mention the PTL tests that go with it).  I'm fine with having the admin jump through some extra hoops until your next pull request.

Reply:
Implemented.